### PR TITLE
[BugFix]Fix some bug about binlog config

### PR DESCRIPTION
**Pull request by smartlxh**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

- fix bug, can not load editlog of `modify_binlog` and `modify binlogAvailableVersion`
- fix bug, if only `binlog_enable` is set , then binlog config can not set successfully
- change the logic of `setBinlogConfig`
- fix the logic of `checkIsPartitionChanged`
- optimize some code, such as the format of log info

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 19 / 23 (82.61%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/binlog/BinlogManager.java|2|6|33.33%|[74, 78, 81, 97]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/leader/ReportHandler.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/binlog/BinlogConfig.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeHandler.java|2|2|100.00%|[]|

**Review by murphyatwork — DISMISSED** [review]

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Comment by smartlxh** [comment]
run starrocks_admit_test

File: be/src/agent/agent_task.cpp [hunk 1]
```diff
@@ -700,8 +700,9 @@ void run_update_meta_info_task(const std::shared_ptr<UpdateTabletMetaInfoAgentTa
             case TTabletMetaType::DISABLE_BINLOG:
                 break;
             case TTabletMetaType::BINLOG_CONFIG:
+                LOG(INFO) << "update tablet:" << tablet->tablet_id()
```
Comment (caneGuy) [inline be/src/agent/agent_task.cpp:703]: update binlog config of tablet from xx to xx

File: fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java [hunk 2]
```diff
@@ -1418,6 +1418,8 @@ public boolean updateBinlogConfigMeta(Database db, Long tableId, Map<String, Str
         newBinlogConfig.incVersion();
         try {
             GlobalStateMgr.getCurrentState().modifyBinlogMeta(db, olapTable, newBinlogConfig);
+            LOG.info("update binlog config of table {}, the binlog after modified is : {}",
```
Comment (caneGuy) [inline fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java:1421]: please print binlog config before modified

File: fe/fe-core/src/main/java/com/starrocks/leader/ReportHandler.java [hunk 3]
```diff
@@ -1283,7 +1283,7 @@ private static void handleSetTabletBinlogConfig(long backendId, Map<Long, TTable
                         tabletToBinlogConfig.add(new ImmutableTriple<>(tabletId, tabletInfo.schema_hash,
                                 olapTable.getCurBinlogConfig()));
                     } else if (beBinlogConfigVersion == feBinlogConfigVersion) {
-                        if (olapTable.enableBinlog() && olapTable.getBinlogAvailableVersion().size() == 0) {
+                        if (olapTable.containsBinlogConfig() && olapTable.getBinlogAvailableVersion().size() == 0) {
```
Comment (caneGuy) [inline fe/fe-core/src/main/java/com/starrocks/leader/ReportHandler.java:1286]: what if we just has binlog_ttl？》

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java [hunk 4]
```diff
@@ -2386,6 +2387,14 @@ private void createOlapOrLakeTable(Database db, CreateTableStmt stmt) throws Ddl
                 } else {
                     throw new DdlException("Unsupported partition method: " + partitionInfo.getType().name());
                 }
+                // if binlog_enable is true when creating table,
+                // then set binlogAvailableVersion without statistics through reportHandler
+                if (olapTable.containsBinlogConfig() && olapTable.enableBinlog()) {
```
Comment (caneGuy) [inline fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:2392]: just check if we have enabled binlog or not?

File: fe/fe-core/src/main/java/com/starrocks/binlog/BinlogManager.java [hunk 5]
```diff
@@ -89,7 +89,14 @@ public void checkAndSetBinlogAvailableVersion(Database db, OlapTable table, long
                 // check if partitions has changed
                 db.readLock();
                 Set<Long> partitions = tableIdToPartitions.computeIfAbsent(table.getId(), key -> new HashSet<>());
-                boolean isPartitionChanged = checkIsPartitionChanged(partitions, table.getAllPartitions());
+                boolean isPartitionChanged = false;
+                // if partitions is empty indicates that the tablet is
+                // the first tablet of the table to be reported,
+                // no need to check whether the partitions have changed
```
Comment (caneGuy) [inline fe/fe-core/src/main/java/com/starrocks/binlog/BinlogManager.java:95]: have changed -> have been changed

File: fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java (hunk 2 above)
Comment (smartlxh) [reply fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java:1421]: the current binlog config has been printed

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 16 / 21 (76.19%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/binlog/BinlogManager.java|2|6|33.33%|[74, 78, 81, 97]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|9|10|90.00%|[2390]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/leader/ReportHandler.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/binlog/BinlogConfig.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeHandler.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 16 / 21 (76.19%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/binlog/BinlogManager.java|2|6|33.33%|[74, 78, 81, 97]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|9|10|90.00%|[2390]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/leader/ReportHandler.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/binlog/BinlogConfig.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeHandler.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 16 / 21 (76.19%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/binlog/BinlogManager.java|2|6|33.33%|[74, 78, 81, 97]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|9|10|90.00%|[2390]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/leader/ReportHandler.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/binlog/BinlogConfig.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeHandler.java|2|2|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java (hunk 4 above)
Comment (smartlxh) [reply fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java]: done

File: fe/fe-core/src/main/java/com/starrocks/leader/ReportHandler.java (hunk 3 above)
Comment (smartlxh) [reply fe/fe-core/src/main/java/com/starrocks/leader/ReportHandler.java]: done

File: fe/fe-core/src/main/java/com/starrocks/binlog/BinlogManager.java (hunk 5 above)
Comment (smartlxh) [reply fe/fe-core/src/main/java/com/starrocks/binlog/BinlogManager.java]: done

File: be/src/agent/agent_task.cpp (hunk 1 above)
Comment (smartlxh) [reply be/src/agent/agent_task.cpp]: ok, the binlog config after modified will be print in tablet_meta#set_binlog_config

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java [hunk 6]
```diff
@@ -2386,6 +2387,14 @@ private void createOlapOrLakeTable(Database db, CreateTableStmt stmt) throws Ddl
                 } else {
                     throw new DdlException("Unsupported partition method: " + partitionInfo.getType().name());
                 }
+                // if binlog_enable is true when creating table,
+                // then set binlogAvailableVersion without statistics through reportHandler
+                if (olapTable.enableBinlog()) {
```
Comment (caneGuy) [inline fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:2392]: enableBinlog rename to isBinlogEnabled

**Review by caneGuy — COMMENTED** [review]
other LGTM

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 17 / 24 (70.83%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/binlog/BinlogManager.java|2|7|28.57%|[74, 78, 81, 97, 272]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|9|11|81.82%|[2390, 4194]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/leader/ReportHandler.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/binlog/BinlogConfig.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeHandler.java|2|2|100.00%|[]|

**Review by murphyatwork — DISMISSED** [review]

**Review by caneGuy — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by smartlxh** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 33 / 41 (80.49%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/binlog/BinlogManager.java|2|7|28.57%|[74, 78, 81, 97, 272]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|11|13|84.62%|[2168, 4194]|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeHandler.java|8|9|88.89%|[1427]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/leader/ReportHandler.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/TableProperty.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/binlog/BinlogConfig.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 33 / 41 (80.49%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/binlog/BinlogManager.java|2|7|28.57%|[74, 78, 81, 97, 272]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|11|13|84.62%|[2168, 4194]|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeHandler.java|8|9|88.89%|[1427]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/leader/ReportHandler.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/TableProperty.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/binlog/BinlogConfig.java|1|1|100.00%|[]|

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Review by murphyatwork — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**